src/soundsourceproxy.cpp

```diff
@@ -273,8 +273,8 @@ SoundSourceProxy::SaveTrackMetadataResult SoundSourceProxy::saveTrackMetadata(
     SoundSourceProxy proxy(pTrack);
     if (!proxy.m_pSoundSource.isNull()) {
         Mixxx::TrackMetadata trackMetadata;
-        bool parsedFromFile = pTrack->getHeaderParsed();
-        pTrack->getMetadata(&trackMetadata);
+        bool parsedFromFile = false;
+        pTrack->getTrackMetadata(&trackMetadata, &parsedFromFile);
```

Comment:
Not sure if here is a good place for CPU optimization, but at least is it possible. 
This function It is called in sometimes in a a loop like in the browse feature, 

How about making trackMetadata a pointer member of pTrack? This would allow us to get around some extra deep copies. We basically need only one deep copy, when creating the metadata clone.
Even the default constructor can be optimized if we return a copy constructed metadata object from    getTrackMetadata()

A separate bool parsedFromFile = pTrack->getHeaderParsed(); would allow us to return without copy the metadata content,